### ISSUE TITLE
router.replace doesn't appear to be working

### ISSUE THREAD

**Issue by CodySwannGT** [expo/router#479]
### Summary

[Given this navigation](https://capture.dropbox.com/hyGDQzIL8fyHmMEs), I want the links to act like bottom tabs.

1. If you are drilled down into "Buy" nav and click "Buy", the app replaces the nav stack with the '/buy' route which is the initial screen for that tab
2. If you are drilled down into the "Sell" nav and click "Buy", the app jumpsTo the Buy tab. If you click back to "Sell" the app takes you to the last screen you were on

However, if I'm drilled down into "Buy" and click "Buy" (which calls `replace('/buy')`, nothing happens

### Minimal reproducible example

````
  const segments = useSegments();
  const currentSegment = segments.slice(-1)[0];
  const isInPath = segments.includes(segment);
  const isCurrent = currentSegment === segment;
  const router = useRouter();

  const handlePress = useCallback(
    () =>
      isCurrent
        ? null
        : isInPath
        ? router.replace(`/${segment}`)
        : navigator.jumpTo(segment),
    [isCurrent, isInPath, router, segment, navigator]
  );
````

**Comment by WKampel** [comment]
Did you ever figure this out?

**Comment by CodySwannGT** [comment]
@WKampel no. I don't think there's a way to make it work with router right now. As a workaround, I used `navigation.popToTop()`
